### PR TITLE
[experiment] Enable ValidationChecker in analyzer in release build

### PR DESCRIPTION
**Pull request by vdimir**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

We have stateless test run in CI for analyzer with release build, let's check that it is not broken

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 3436be801965cba4d747ffbf22937e4c11787529 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56865/3436be801965cba4d747ffbf22937e4c11787529/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6893325173">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56865/3436be801965cba4d747ffbf22937e4c11787529/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Core/Settings.h [hunk 1]
```diff
@@ -349,7 +349,7 @@ class IColumn;
     M(Float, opentelemetry_start_trace_probability, 0., "Probability to start an OpenTelemetry trace for an incoming query.", 0) \
     M(Bool, opentelemetry_trace_processors, false, "Collect OpenTelemetry spans for processors.", 0) \
     M(Bool, prefer_column_name_to_alias, false, "Prefer using column names instead of aliases if possible.", 0) \
-    M(Bool, allow_experimental_analyzer, false, "Allow experimental analyzer", 0) \
+    M(Bool, allow_experimental_analyzer, true, "Allow experimental analyzer", 0) \
```
Comment (vdimir) [inline src/Core/Settings.h:352]:
```suggestion
    M(Bool, allow_experimental_analyzer, false, "Allow experimental analyzer", 0) \
```

**Comment by vdimir** [comment]
> Stateless tests (release, analyzer) — fail: 3, passed: 5838, skipped: 11 [Details](https://s3.amazonaws.com/clickhouse-test-reports/56865/7497495f05beb22c871045b500a8c4ed9e788fda/stateless_tests__release__analyzer_.html)

In one test check failed:
```
2023-11-16 17:21:24 02662_first_last_value:                                                 [ FAIL ] 0.09 sec. - return code: 49
2023-11-16 17:21:24 [f3dcc5c437f7] 2023.11.16 14:21:24.661852 [ 6847 ] {2810c5d6-625c-42ec-a5a5-a2e124de96ad} <Error> executeQuery: Code: 49. DB::Exception: Function any(test_nocdgf9p.test.b) expects 1 argument to have Int32 type but receives Nullable(Int32) after running QueryAnalysis pass. (LOGICAL_ERROR) (version 23.11.1.1183) (from [::1]:40922) (comment: 02662_first_last_value.sql) (in query: select first_value(b) respect nulls from test;), Stack trace (when copying this message, always include the lines below):
```

**Comment by novikd** [comment]
I think we need to replace  `Stateless tests (release, analyzer)` with debug and split it into several parts.

**Pull request closed without merging**